src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,622 @@
+				}
+			}
+		} else {
+			throw new NotImplemented(`Iteration over WebUntis data must be teachers or rooms.`);
+		}
+
+		const result = [];
+
+		// Collect times for each course
+		// TODO: filter time slots for non-existing Schul-Cloud classes?
+		for (let course of intermediateData.courses) {
+			const times = [];
+			for (const timetableEntry of course.timetable) {
+				const newEntry = {
+					weekday: this.getWeekDay(timetableEntry.date),
+					startTime: this.getStartTime(timetableEntry.startTime),
+					duration: this.getDuration(timetableEntry.startTime, timetableEntry.endTime),
+					room: timetableEntry.room,
+					count: 1,
+				};
+				
+				let entryFound = false;
+				for (const givenEntry of times) {
+					if (givenEntry.weekday === newEntry.weekday
+						&& givenEntry.startTime === newEntry.startTime
+						&& givenEntry.duration === newEntry.duration
+						&& givenEntry.room === newEntry.room) {
+						givenEntry.count += 1;
+						entryFound = true;
+					}
+				}
+				
+				if (!entryFound) {
+					times.push(newEntry);
+				}
+			}
+
+			// Assumption: After 2 entries it is recurring
+			const filteredTimes = times.filter(entry => entry.count >= 2);
+
+			result.push({
+				teacher: course.teacher,
+				classes: course.classes,
+				subject: course.subject,
+				times: filteredTimes,
+				state: 'new'
+			});
+		}
+
+		return result;
+	}
+
+	/**
+	* 
+	* @param {*} metaData 
+	*/
+	async fetchData(metaData) {
+		// Not required as all information are fetched during phase 1
+
+		return metaData;
+	}
+
+	/**
+	* 
+	* @param {*} metadata 
+	*/
+	async storeMetadata(metaData, params) {
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		await Promise.all(metaData.map(async (entry) => {
+			const metadataResults = await webUntisMetadataService.find({ query: {
+				datasourceId: params.datasourceId,
+				teacher: entry.teacher,
+				classes: entry.classes,
+				subject: entry.subject
+			}, paginate: false });
+			const result = metadataResults[0];
+
+			if (result !== undefined) { // patch/replace existing
+				await webUntisMetadataService.patch(result._id, { state: entry.state, times: entry.times });
+			} else { // create new
+				await webUntisMetadataService.create(Object.assign(
+					{ datasourceId: params.datasourceId },
+					entry
+				));
+			}
+		}));
+	}
+
+	/**
+	* 
+	* @param {*} data 
+	*/
+	async createCourses(config, data, params) {
+		// Convert metadata to actual schulcloud courses
+		// And reflect changes in metadata store
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		const importCondition = this.getImportConditionEvaluator(params);
+
+		for (let entry of data) {
+			if (importCondition(entry._id.toString())) {
+				await this.obtainAndUpdateCourseAndClass(config, entry)
+					.then(() => webUntisMetadataService.patch(entry._id, { state: 'imported' }))
+					// .catch(() => webUntisMetadataService.patch(entry._id, { state: 'errored' }));
+					;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 tabs but found 5.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)